extra/mariabackup/xtrabackup.cc

```diff
@@ -3486,6 +3484,8 @@ static bool xtrabackup_copy_mmap_logfile()
 
       retry_count= 0;
     }
+    else if (r == recv_sys_t::GOT_EOF)
+      break;
     else
     {
```

Comment:
The `else` branch below the added `else if` should be unreachable code and be removed. In fact, also the `else if` can be removed. Furthermore, it would seem that the `for` loop and the local variable `retry_count` will need to be removed. That is, we’d only have the `do` loop inside the initial `if`.